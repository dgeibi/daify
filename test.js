/* eslint-env jest */

const { arrayify, dictify } = require('./src')

test('just works', () => {
  const opts = {
    by: 'id',
    include: {
      lists: 'id',
      'namespace.inners': 'id',
    },
  }

  const getData = () => [
    { id: 1, hello: 1, lists: [{ id: 1, hello: 2 }] },
    { id: 2, hello: 1, lists: [{ id: 1, hello: 2 }] },
    {
      id: 3,
      hello: 1,
      lists: [{ id: 1, hello: 2 }],
      namespace: {
        inners: [{ id: 1, hello: 2 }, { id: 2, hello: 2 }, { id: 3, hello: 2 }],
      },
    },
  ]

  expect(dictify(getData(), opts)).toMatchSnapshot()
  expect(arrayify(dictify(getData(), opts), opts)).toEqual(getData())
})

test('inner key', () => {
  const opts = 'ns.id'

  const getData = () => [
    { ns: { id: 1 } },
    { ns: { id: 2 } },
    { ns: { id: 3 }, yes: true },
  ]
  expect(dictify(getData(), opts)).toMatchSnapshot()
  expect(arrayify(dictify(getData(), opts), opts)).toEqual(getData())
})

test('set key by function', () => {
  const opts = {
    by: x => x.ns.id,
  }

  const getData = () => [
    { ns: { id: 1 } },
    { ns: { id: 2 } },
    { ns: { id: 3 }, yes: true },
  ]
  expect(dictify(getData(), opts)).toMatchSnapshot()
  expect(arrayify(dictify(getData(), opts), opts)).toEqual(getData())
})

test('nested include', () => {
  const opts = {
    by: 'id',
    include: {
      lists: {
        by: 'id',
        include: {
          cards: 'id',
        },
      },
    },
  }

  const getData = () => [
    {
      id: 1,
      lists: [
        {
          id: 1,
          hello: 2,
          cards: [{ id: 99 }, { id: 101 }, { id: 111 }],
        },
      ],
    },
    { id: 2, lists: [{ id: 1, hello: 2 }] },
  ]

  expect(dictify(getData(), opts)).toMatchSnapshot()
  expect(arrayify(dictify(getData(), opts), opts)).toEqual(getData())
})
