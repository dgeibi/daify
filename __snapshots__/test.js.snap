// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inner key 1`] = `
Object {
  "1": Object {
    "ns": Object {
      "id": 1,
    },
  },
  "2": Object {
    "ns": Object {
      "id": 2,
    },
  },
  "3": Object {
    "ns": Object {
      "id": 3,
    },
    "yes": true,
  },
}
`;

exports[`just works 1`] = `
Object {
  "1": Object {
    "hello": 1,
    "id": 1,
    "lists": Object {
      "1": Object {
        "hello": 2,
        "id": 1,
      },
    },
  },
  "2": Object {
    "hello": 1,
    "id": 2,
    "lists": Object {
      "1": Object {
        "hello": 2,
        "id": 1,
      },
    },
  },
  "3": Object {
    "hello": 1,
    "id": 3,
    "lists": Object {
      "1": Object {
        "hello": 2,
        "id": 1,
      },
    },
    "namespace": Object {
      "inners": Object {
        "1": Object {
          "hello": 2,
          "id": 1,
        },
        "2": Object {
          "hello": 2,
          "id": 2,
        },
        "3": Object {
          "hello": 2,
          "id": 3,
        },
      },
    },
  },
}
`;

exports[`nested include 1`] = `
Object {
  "1": Object {
    "id": 1,
    "lists": Object {
      "1": Object {
        "cards": Object {
          "101": Object {
            "id": 101,
          },
          "111": Object {
            "id": 111,
          },
          "99": Object {
            "id": 99,
          },
        },
        "hello": 2,
        "id": 1,
      },
    },
  },
  "2": Object {
    "id": 2,
    "lists": Object {
      "1": Object {
        "hello": 2,
        "id": 1,
      },
    },
  },
}
`;

exports[`set key by function 1`] = `
Object {
  "1": Object {
    "ns": Object {
      "id": 1,
    },
  },
  "2": Object {
    "ns": Object {
      "id": 2,
    },
  },
  "3": Object {
    "ns": Object {
      "id": 3,
    },
    "yes": true,
  },
}
`;
